pkg/deploymanager/config.go

```diff
@@ -1,133 +1,63 @@
-func (t *DeployManager) GetParameterCodec() runtime.ParameterCodec {
-	return t.parameterCodec
-}
-
-// GetLvmClient is the function used to retrieve the lvm client.
-func (t *DeployManager) GetLvmClient() *rest.RESTClient {
-	return t.lvmClient
-}
-
-func getKubeconfig(kubeconfig string) (*rest.Config, error) {
+// NewDeployManager creates a DeployManager struct with default configuration
+func NewDeployManager() (*DeployManager, error) {
+	kubeconfig := os.Getenv("KUBECONFIG")
 	var config *rest.Config
 	var err error
 	if kubeconfig != "" {
 		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
 	} else {
 		config, err = rest.InClusterConfig()
 	}
-	if err != nil {
-		return config, nil
-	}
-	return config, err
-}
-
-// NewDeployManager creates a DeployManager struct with default configuration.
-func NewDeployManager() (*DeployManager, error) {
-	codecs := serializer.NewCodecFactory(scheme.Scheme)
-	parameterCodec := runtime.NewParameterCodec(scheme.Scheme)
-	var config *rest.Config
-	var lvmConfig *rest.Config
-	var olmConfig *rest.Config
-	var err error
-
-	kubeconfig := os.Getenv("KUBECONFIG")
-	config, err = getKubeconfig(kubeconfig)
-	if err != nil {
-		return nil, err
-	}
-	config.NegotiatedSerializer = serializer.WithoutConversionCodecFactory{CodecFactory: codecs}
-	config.APIPath = "/apis"
-	config.ContentType = runtime.ContentTypeJSON
-	k8sClient, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, err
-	}
-
-	// lvm Operator rest client
-	lvmConfig, err = getKubeconfig(kubeconfig)
-	if err != nil {
-		return nil, err
-	}
-	lvmConfig.GroupVersion = &lvmv1.GroupVersion
-	lvmConfig.NegotiatedSerializer = serializer.WithoutConversionCodecFactory{CodecFactory: codecs}
-	lvmConfig.APIPath = "/apis"
-	lvmConfig.ContentType = runtime.ContentTypeJSON
-	if lvmConfig.UserAgent == "" {
-		lvmConfig.UserAgent = rest.DefaultKubernetesUserAgent()
-	}
-	lvmClient, err := rest.RESTClientFor(lvmConfig)
-	if err != nil {
-		return nil, err
-	}
-
-	// controller-runtime client
-	myScheme := runtime.NewScheme()
-	utilruntime.Must(lvmv1.AddToScheme(myScheme))
-	utilruntime.Must(scheme.AddToScheme(myScheme))
-	crClient, err := crclient.New(config, crclient.Options{Scheme: myScheme})
 	if err != nil {
 		return nil, err
 	}
 
-	// olm client
-	olmConfig, err = getKubeconfig(kubeconfig)
-	if err != nil {
-		return nil, err
-	}
-	olmConfig.NegotiatedSerializer = serializer.WithoutConversionCodecFactory{CodecFactory: scheme.Codecs}
-	olmConfig.APIPath = "/apis"
-	olmConfig.ContentType = runtime.ContentTypeJSON
-	olmClient, err := olmclient.NewForConfig(olmConfig)
+	crClient, err := crclient.New(config, crclient.Options{Scheme: scheme})
 	if err != nil {
 		return nil, err
 	}
 
 	return &DeployManager{
-		olmClient:      olmClient,
-		k8sClient:      k8sClient,
-		crClient:       crClient,
-		lvmClient:      lvmClient,
-		parameterCodec: parameterCodec,
+		crClient: crClient,
+		Ctx:      context.TODO(),
```

Comment:
Please remove the ctx from the deploymanager. Lets stick to context.TODO for now.